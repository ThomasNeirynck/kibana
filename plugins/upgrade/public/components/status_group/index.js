export { StatusGroup } from './status_group';
