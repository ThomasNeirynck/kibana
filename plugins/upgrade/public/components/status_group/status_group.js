import React, { Component } from 'react';
import classNames from 'classnames';


export class StatusGroup extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    className: React.PropTypes.string,
    isInitiallyCollapsed: React.PropTypes.bool,
    status: React.PropTypes.node,
    title: React.PropTypes.node,
  }

  static defaultProps = {
    children: null,
    className: null,
    isInitiallyCollapsed: false,
    status: null,
    title: null,
  }

  constructor(props) {
    super(props);

    this.state = {
      isCollapsed: props.isInitiallyCollapsed,
    };
  }

  toggleIsCollapsed = () => {
    this.setState(({ isCollapsed }) => ({
      isCollapsed: !isCollapsed,
    }));
  }

  render() {
    const { children, className, status, title } = this.props;
    const { isCollapsed } = this.state;

    const toggleIconClasses = classNames('kuiIcon', 'kuiToggleButton__icon', {
      'fa-caret-right': isCollapsed,
      'fa-caret-down': !isCollapsed,
    });

    return (
      <div className={ className }>
        <div className="kuiTogglePanelHeader">
          <button className="kuiToggleButton" onClick={ this.toggleIsCollapsed }>
            <span className="upgradePluginPanelHeader">
              <span>
                <span className={ toggleIconClasses } />
                { title }
              </span>
              <span>
                { status }
              </span>
            </span>
          </button>
        </div>
        {
          isCollapsed
            ? null
            : <div className="kuiTogglePanelContent">{children}</div>
        }
      </div>
    );
  }
}
