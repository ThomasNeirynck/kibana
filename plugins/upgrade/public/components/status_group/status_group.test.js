import { shallow } from 'enzyme';
import React from 'react';

import { StatusGroup } from './status_group';


describe('StatusGroup', () => {
  test('renders the className, title and status properties', () => {
    const component = (
      <StatusGroup
        className="specificStatusGroup"
        title="A different title"
        status="the status"
      />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders its children by default', () => {
    const component = (
      <StatusGroup>
        Info Message
      </StatusGroup>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders its children by default when isInitiallyCollapsed is false', () => {
    const component = (
      <StatusGroup isInitiallyCollapsed={ false }>
        Info Message
      </StatusGroup>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('does not render its children by default when isInitiallyCollapsed is true', () => {
    const component = (
      <StatusGroup isInitiallyCollapsed={ true }>
        Info Message
      </StatusGroup>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders its children after being expanded via the toggle button', () => {
    const component = (
      <StatusGroup isInitiallyCollapsed={ true }>
        Info Message
      </StatusGroup>
    );

    const wrapper = shallow(component);
    wrapper.find('button').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  test('does not render its children after being collapsed via the toggle button', () => {
    const component = (
      <StatusGroup isInitiallyCollapsed={ false }>
        Info Message
      </StatusGroup>
    );

    const wrapper = shallow(component);
    wrapper.find('button').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });
});
