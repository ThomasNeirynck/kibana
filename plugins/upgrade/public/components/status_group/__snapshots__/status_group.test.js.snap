// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusGroup does not render its children after being collapsed via the toggle button 1`] = `
<div
  className={null}
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-right"
          />
        </span>
        <span />
      </span>
    </button>
  </div>
</div>
`;

exports[`StatusGroup does not render its children by default when isInitiallyCollapsed is true 1`] = `
<div
  className={null}
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-right"
          />
        </span>
        <span />
      </span>
    </button>
  </div>
</div>
`;

exports[`StatusGroup renders its children after being expanded via the toggle button 1`] = `
<div
  className={null}
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-down"
          />
        </span>
        <span />
      </span>
    </button>
  </div>
  <div
    className="kuiTogglePanelContent"
  >
    Info Message
  </div>
</div>
`;

exports[`StatusGroup renders its children by default 1`] = `
<div
  className={null}
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-down"
          />
        </span>
        <span />
      </span>
    </button>
  </div>
  <div
    className="kuiTogglePanelContent"
  >
    Info Message
  </div>
</div>
`;

exports[`StatusGroup renders its children by default when isInitiallyCollapsed is false 1`] = `
<div
  className={null}
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-down"
          />
        </span>
        <span />
      </span>
    </button>
  </div>
  <div
    className="kuiTogglePanelContent"
  >
    Info Message
  </div>
</div>
`;

exports[`StatusGroup renders the className, title and status properties 1`] = `
<div
  className="specificStatusGroup"
>
  <div
    className="kuiTogglePanelHeader"
  >
    <button
      className="kuiToggleButton"
      onClick={[Function]}
    >
      <span
        className="upgradePluginPanelHeader"
      >
        <span>
          <span
            className="kuiIcon kuiToggleButton__icon fa-caret-down"
          />
          A different title
        </span>
        <span>
          the status
        </span>
      </span>
    </button>
  </div>
  <div
    className="kuiTogglePanelContent"
  />
</div>
`;
